src/Chapter2.hs

```diff
@@ -605,9 +636,18 @@ Implement a function that duplicates each element of the list
 "aabbaacc"
 
 -}
+-- Solution 1
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate l = go [] (reverse l)
+  where
+    go :: [a] -> [a] -> [a]
+    go result [] = result
+    go result (x:xs) = go (x : x : result) xs
 
+-- Solution 2, took some time to think about this solution
```

Comment:
Nice one! 

This solution has a number of advantages compared to the previous one: it is more efficient and it is also streaming (the previous one won't work on infinite lists).
Well done 👍🏼